engine/channels.go

```diff
@@ -49,6 +76,60 @@ const (
 	ProvideReceiptsByBlockID = RequestReceiptsByBlockID
 )
 
+// initializeTopicMap initializes an instance of topicMap and populates it with the topics and their
+// corresponding list of roles.
+func initializeTopicMap() {
+	topicMap = make(map[string]flow.RoleList)
+
+	// Channels for test
+	topicMap[TestNetwork] = flow.RoleList{flow.RoleCollection, flow.RoleConsensus, flow.RoleExecution,
+		flow.RoleVerification, flow.RoleAccess}
+	topicMap[TestMetrics] = flow.RoleList{flow.RoleCollection, flow.RoleConsensus, flow.RoleExecution,
+		flow.RoleVerification, flow.RoleAccess}
+
+	// Channels for consensus protocols
+	topicMap[ConsensusCommittee] = flow.RoleList{flow.RoleConsensus}
+
+	// Channels for protocols actively synchronizing state across nodes
+	topicMap[SyncCommittee] = flow.RoleList{flow.RoleConsensus}
+	topicMap[SyncExecution] = flow.RoleList{flow.RoleExecution}
+
+	// Channels for actively pushing entities to subscribers
+	topicMap[PushTransactions] = flow.RoleList{flow.RoleCollection}
+	topicMap[PushGuarantees] = flow.RoleList{flow.RoleCollection, flow.RoleConsensus}
+	topicMap[PushBlocks] = flow.RoleList{flow.RoleCollection, flow.RoleConsensus, flow.RoleExecution,
+		flow.RoleVerification, flow.RoleAccess}
+	topicMap[PushReceipts] = flow.RoleList{flow.RoleConsensus, flow.RoleExecution, flow.RoleVerification,
+		flow.RoleAccess}
+	topicMap[PushApprovals] = flow.RoleList{flow.RoleConsensus, flow.RoleVerification}
+
+	// Channels for actively requesting missing entities
+	topicMap[RequestCollections] = flow.RoleList{flow.RoleCollection, flow.RoleExecution}
```

Comment:
this should have had `flow.RoleAccess`, since the access nodes also request for Collections (after they get a block)